modules/gke-cluster-autopilot/variables.tf

```diff
@@ -216,9 +216,34 @@ variable "service_account" {
 }
 
 variable "tags" {
-  description = "Network tags applied to nodes."
+  description = "Applies the specified network tags to all nodes in this Autopilot cluster. Network tags are metadata on Compute Engine virtual machines (VMs) that allow you to make firewall rules and routes applicable to specific VM instances. In GKE, you can use network tags to make firewall rules or routes applicable to the nodes in your cluster."
   type        = list(string)
-  default     = null
+  default     = []
+  nullable    = false
+  validation {
```

Comment:
We tend to only validate in specific cases, for example when a string field is really an enum, or when an object configures a resource via top-level attributes and only one is needed (e.g. healthchecks).

As a further example of our approach, we tend not to validate cases where the API returns a clear error. For example with disallowed characters in strings, length of strings, etc.

While I'm not entirely against this approach (it does make the variable files harder to parse though), it's way overkill vs our standard practices, and if we decide to follow this path we need to do it consistently across all modules.

I would roll this back. :)

Reply:
Validation was removed as requested. Conversation resolved :shipit: